### ISSUE TITLE
Using native.new_http_archive in WORKSPACE causes ctx.workspace_name to be an empty string

### ISSUE THREAD

**Issue by pspeter3** [bazelbuild/bazel#1213]

**Comment by damienmg** [comment]
Can we get a bit more information?

  - `bazel version`
  - Precise reproduction.

**Comment by pspeter3** [comment]
I just ran this with the following files under the `0.2.2b` and the issue did not repeat. Given that, I am closing this bug and will upgrade to the latest version.

#### WORKSPACE

```py
workspace(
    name = "demo"
)

load("//:sass.bzl", "sass_repositories")

sass_repositories()
```

#### BUILD

```py
load("//:test.bzl", "test")

test(
    name = "demo"
)
```

#### test.bzl

```py
def _test_impl(ctx):
    print(ctx.workspace_name)

test = rule(
    implementation = _test_impl
)
```

#### sass.bzl

```py
LIBSASS_BUILD_FILE = """
package(default_visibility = ["@sassc//:__pkg__"])

filegroup(
    name = "srcs",
    srcs = glob([
         "src/**/*.h*",
         "src/**/*.c*",
    ]),
)

# Includes directive may seem unnecessary here, but its needed for the weird
# interplay between libsass/sassc projects. This is intentional.
cc_library(
    name = "headers",
    includes = ["include"],
    hdrs = glob(["include/**/*.h"]),
)
"""

SASSC_BUILD_FILE = """
package(default_visibility = ["//visibility:public"])

cc_binary(
    name = "sassc",
    srcs = [
        "@libsass//:srcs",
        "sassc.c",
        "sassc_version.h",
],
    linkopts = ["-ldl", "-lm"],
    deps = ["@libsass//:headers"],
)
"""

def sass_repositories():
  native.new_http_archive(
      name = "libsass",
      url = "https://github.com/sass/libsass/archive/3.3.0-beta1.tar.gz",
      sha256 = "6a4da39cc0b585f7a6ee660dc522916f0f417c890c3c0ac7ebbf6a85a16d220f",
      build_file_content = LIBSASS_BUILD_FILE,
      strip_prefix = "libsass-3.3.0-beta1",
  )

  native.new_http_archive(
      name = "sassc",
      url = "https://github.com/sass/sassc/archive/3.3.0-beta1.tar.gz",
      sha256 = "87494218eea2441a7a24b40f227330877dbba75c5fa9014ac6188711baed53f6",
      build_file_content = SASSC_BUILD_FILE,
      strip_prefix = "sassc-3.3.0-beta1",
  )
```

**Comment by damienmg** [comment]
Oh thanks, yes you were hitting a bug that got fixed in 0.2.2
